BootloaderCommonPkg/Library/FullMemoryAllocationLib/FullMemoryAllocationLib.c

```diff
@@ -632,4 +632,36 @@ AllocateRuntimePool (
   )
 {
   return InternalAllocatePool (EfiRuntimeServicesData, AllocationSize);
+}
+
+/**
+  This function allocates temporary memory pool.
+
+  @param[in]  AllocationSize    The memory pool size to allocate.
+
+  @retval     A pointer to the allocated temporary buffer or NULL if allocation fails.
+
+**/
+VOID *
+EFIAPI
+AllocateTemporaryMemory (
```

Comment:
Comparing with AllocateRuntimePool (), better to rename it to AllocateBootTimePool () or AllocateTemporaryPool ()?  with Pool in the name it is clear that AllocationSize is in byte unit.

Reply:
AllocateTemporaryMemory is paired with FreeTemporaryMemory .  
It think it is clear that it is byte.   But if you insist in  AllocateTemporaryPool(), I can also make the change.

Reply:
OK. It is fine to use it instead of changing FreeTemporaryMemory()